src/app/services/filter-product.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { ProductWrapper } from '../classes/product';
+
+@Injectable()
+export class FilterProductService {
+  private goods: ProductWrapper[] = [];
+  private filter: any = new Map();
+
+  addGoods(productArr:ProductWrapper[]){
+    this.goods = productArr;
+  }
+
+  goodsFilter(type: string, value: string ): ProductWrapper[] {
+
+    if (value === 'ALL') {
+      this.filter.delete(type);
+    } else {
+      this.filter.set(type, value);
+    }
+
+    let filteredGoods: ProductWrapper[] = this.goods;
+
+    if (this.filter.length !== 0) {
```

Comment:
Doens't have sense to check length according to for for empty array doesn't throw an error